src/colorlibrary.cpp

```diff
@@ -1,18 +1,43 @@
 #include "colorlibrary.h"
 
+const static QVector<QColor> DefaultColors =
+{
+    QColor("#f2ec99"),
+    QColor("#bfe4f3"),
+    QColor("#bfe968"),
+    QColor("#c6caf8"),
+    QColor("#eec757"),
+    QColor("#9994f1"),
+    QColor("#edb0e8"),
+    QColor("#cebff3"),
+    QColor("#a5f0c6"),
+    QColor("#f5b0b4")
+};
+
 ColorLibrary::ColorLibrary()
 {
     m_colorLibrary = new QQueue<QColor>();
-    m_colorLibrary->enqueue(QColor("#f2ec99"));
-    m_colorLibrary->enqueue(QColor("#bfe4f3"));
-    m_colorLibrary->enqueue(QColor("#bfe968"));
-    m_colorLibrary->enqueue(QColor("#c6caf8"));
-    m_colorLibrary->enqueue(QColor("#eec757"));
-    m_colorLibrary->enqueue(QColor("#9994f1"));
-    m_colorLibrary->enqueue(QColor("#edb0e8"));
-    m_colorLibrary->enqueue(QColor("#cebff3"));
-    m_colorLibrary->enqueue(QColor("#a5f0c6"));
-    m_colorLibrary->enqueue(QColor("#f5b0b4"));
+    AddColors(DefaultColors);
+}
+
+ColorLibrary::ColorLibrary(QVector<QColor> usedColors)
+{
+    m_colorLibrary = new QQueue<QColor>();
+    QVector<QColor> colorsToAdd;
+    for(const auto& color : DefaultColors)
+    {
+        if (!usedColors.contains(color))
+            colorsToAdd.push_back(color);
+    }
+    AddColors(colorsToAdd);
```

Comment:
The reason I had hand picked those ~10 default colors is because I thought they had good enough contrast to be on the same logtab at the same time. I don't think that manually adding more and more default colors is a good long term approach (and there would definitely be diminishing returns on each default color we add compared to the effort of making sure each new one does not conflict).

However, automatically generating good default colors is tricky due to possible conflicts, and I don't think it is worth the time investing in a more complex color generation algorithm.

A simple short term fix could be "If we don't have any default colors left, randomly generate colors". Not ideal, but I think the amount of times a user will exceed the default filters will be relatively small?